index.js

```diff
@@ -62,6 +62,56 @@ function validateList (list, supported, name) {
   return list
 }
 
+function getNameAndVersion (opts, cb) {
+  // Use provided name and version
+  if (opts.name && opts.version) return cb(null)
+
+  function searchUpwards (name, dir, fn, cb) {
+    var filename = path.resolve(path.join(dir, name))
+    fs.readFile(filename, function (err, file) {
+      // Error accessing file, but ENOENT is okay, ignore that
+      if (err && err.code !== 'ENOENT') return cb(err)
+      else if (!err) {
+        try {
+          var pkg = JSON.parse(file)
+          if (fn(pkg, dir)) return cb(null)
+        } catch (err) {
+          err.message = 'Error parsing ' + filename + ': ' + err.message
+          return cb(err)
+        }
+      }
+      var parent = path.join(dir, '..')
+      if (path.resolve(parent) === path.resolve(dir)) return cb(null)
+      searchUpwards(name, parent, fn, cb)
+    })
+  }
```

Comment:
I would prefer to use https://www.npmjs.com/package/resolve instead of this function